src/controllers/cart.controller.ts

```diff
@@ -0,0 +1,218 @@
+// src/cart/cart.controller.ts
+
+import type { Prisma } from '@prisma/client';
+import type { NextFunction, Request, Response } from 'express';
+
+import type { PatchCartInput } from '../dtos/cart.dto.js';
+import { getCartItemWithDetails, getCartWithItems } from '../repositories/cart.repository.js';
+import * as cartService from '../services/cart.service.js';
+
+type SizeDetail = { ko?: string; en?: string; [key: string]: unknown };
+
+const toSizeDetail = (value: Prisma.JsonValue | null): SizeDetail | null => {
+  if (!value || typeof value !== 'object' || Array.isArray(value)) return null;
+  return value as SizeDetail;
+};
+
+// 공통 401 응답 헬퍼 (선택이지만 편해서 추가)
+function getUserIdOr401(req: Request, res: Response): string | null {
+  const user = req.user;
+  if (!user) {
+    res.status(401).json({
+      statusCode: 401,
+      message: '인증이 필요합니다.',
+      error: 'Unauthorized',
+    });
+    return null;
+  }
+  return user.id;
+}
+
+// POST /api/cart : 장바구니 생성 (또는 기존 카트 반환)
+export async function createCart(req: Request, res: Response) {
+  const userId = getUserIdOr401(req, res);
+  if (!userId) return; // 위에서 401 응답 후 종료
+
+  const cart = await cartService.createCart(userId);
+  if (!cart) {
+    return res.status(500).json({
```

Comment:
```suggestion
    return res.status(STATUS_CODE.INTERNAL_SERVER_ERROR).json({
```
에러 코드의 경우에 `constant.ts` 파일에 정리해둬서 상수를 호출해서 사용하면 나중에 유지보수할 때 좋을 것 같아요